File: app/channels/import_channel.rb
class ImportChannel < ApplicationCable::Channel
  def subscribed
    stream_from "update_products"
  end

  def receive(data)
    puts data["message"]
    ActionCable.server.broadcast("test", "ActionCable is connected")
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end

File: app/javascript/channels/import_channel.js
import consumer from "channels/consumer"

consumer.subscriptions.create("ImportChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    this.send({ message: 'Client is live' })
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log(data)
  }
});
